File: app/src/main/java/com/prosabdev/fluidmusic/ui/custom/CustomShapeableImageViewImageViewRatio11.java
package com.prosabdev.fluidmusic.ui.custom;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.Nullable;

import com.google.android.material.imageview.ShapeableImageView;

public class CustomShapeableImageViewImageViewRatio11 extends ShapeableImageView {
    public CustomShapeableImageViewImageViewRatio11(Context context) {
        super(context);
    }

    public CustomShapeableImageViewImageViewRatio11(Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomShapeableImageViewImageViewRatio11(Context context, @Nullable AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        int width = getMeasuredWidth();
        int height = getMeasuredHeight(); // Optimization so we don't measure twice unless we need to
        if (width != height) {
            setMeasuredDimension(width, width);
        }
    }


}

File: app/src/main/java/com/prosabdev/fluidmusic/ui/custom/CustomShapeableImageViewImageViewRatio11.kt
package com.prosabdev.fluidmusic.ui.custom

import android.content.Context
import android.util.AttributeSet
import com.google.android.material.imageview.ShapeableImageView

class CustomShapeableImageViewImageViewRatio11 : ShapeableImageView {
    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyle: Int) : super(
        context,
        attrs,
        defStyle
    )

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        val width = measuredWidth
        val height = measuredHeight // Optimization so we don't measure twice unless we need to
        if (width != height) {
            setMeasuredDimension(width, width)
        }
    }
}
